packages/main/src/models/conversation.ts

```diff
@@ -109,14 +121,20 @@ export function getPresentableElements({
         isRead: cache
           .getFlags(latestEdit ? latestEdit.revision.message.id : message.id)
           .includes("\\Seen"),
-        isStarred: cache.getFlags(message.id).includes("\\Flagged"),
+        isStarred: resources.some(r =>
+          cache.getFlags(r.revision.message.id).includes("\\Flagged")
+        ),
         contents: resources
           .valueSeq()
           .map(getPresentableContent)
           .toArray(),
         date: message.date,
         from: cache.getParticipants(message.id, "from")[0],
         editedAt: latestEdit && latestEdit.revision.message.date,
+        revisions: resources
+          .valueSeq()
+          .flatMap(r => [r.revision.message])
+          .toArray(),
```

Comment:
Good stuff. But I realize that although we only need messages at the moment, it would be highly useful for the future if the `revisions` property includes the entire revision object - both message and part.

Also this use of `flatMap` could be simplified to `.map(r => r.revision.message)`. Calling `xs.flatMap(fn)` is the same as calling `xs.map(fn).flatten()`. In this case after `valueSeq` you get a sequence that looks roughly like this:

```ts
[{ message: m1, part: p1 }, { message: m2, part: p2 }]
```

If we think if the `flatMap` call as two steps (`map` and `flatten`) then after the `map` your sequence looks like this:

```ts
[[m1], [m2]]
```

`flatten` turns a list of lists into one flat list; so after `flatten` the result looks like this:

```ts
[m1, m2]
```

The only reason that flattening is necessary is because you wrap each message in a new array in the `flatMap` callback. If you skip the array wrapping you can use `map` directly and skip a step.